pkg/registry/apis/dashboard/legacy/sql_dashboards.go

```diff
@@ -339,14 +338,9 @@ func (a *dashboardSqlAccess) DeleteDashboard(ctx context.Context, orgId int64, u
 		return dash, false, err
 	}
 
-	id := dash.Spec.GetNestedInt64("id")
-	if id == 0 {
-		return nil, false, fmt.Errorf("could not find id in saved body")
-	}
-
 	err = a.dashStore.DeleteDashboard(ctx, &dashboards.DeleteDashboardCommand{
 		OrgID: orgId,
-		ID:    id,
+		UID:   uid,
```

Comment:
When this function was originally written, you couldn't delete dashboards by uid. That has since been added to the dashboard command